include/db/postgres.inc.php

```diff
@@ -89,7 +89,17 @@ function serendipity_db_reconnect() {
  * @return  string   output string
  */
 function serendipity_db_escape_string($string) {
-    return pg_escape_string($string);
+    global $serendipity;
+
+    if (is_null($string))
+        return $string;
+
+    if (PHP_MAJOR_VERSION < 8 || (PHP_MAJOR_VERSION == 8 && PHP_MINOR_VERSION < 1))
+        # Last supported version is PHP 8.0
+        return pg_escape_string($string);
```

Comment:
What do you think about this instead (untested)?

```
if (version_compare(PHP_VERSION, '8.1.0', '<')) {
    # Versions before 8.1 give the connection by default
    return pg_escape_string($string);
}
```

That's how the core would usually express that expression, and I think it's a bit clearer, isn't it?